src/pkg/gc/gc.go

```diff
@@ -230,36 +228,65 @@ func getRunningContainers() []docker.APIContainers {
 	return running
 }
 
-func removeImagesInBatch(keepLast time.Duration) int {
+func removeImagesInBatch(policy GCPolicy) int {
 	dataMap := getImages()
 
-	dates := helpers.SortDataMap(dataMap)
-	var batch []int64
-	if len(dates) > BatchSizeToDelete {
-		batch = dates[len(dates)-BatchSizeToDelete:]
-	} else {
-		batch = dates
-	}
+	totalDeletedImages := 0
+	batchCounter := 1
 
-	// Create a new map with only the values in the batch
-	batchDataMap := map[int64][]string{}
-	for _, date := range batch {
-		//Notice this might not be exactly BatchSizeToDelete because there might multiple images created at same exact moment
-		batchDataMap[date] = dataMap[date]
+	dates := helpers.SortDataMapReverse(dataMap)
+
+	usedDiskSpace, diskErr := diskSpaceFetcher.GetUsedDiskSpaceInPercents()
+	if diskErr != nil {
+		log.WithField("error", diskErr).Error("Reading disk space failed")
+		return 0
 	}
 
+	for usedDiskSpace > policy.LowDiskSpaceThreshold {
+		var batch []int64
+		if len(dates) > BatchSizeToDelete {
+			// Two pointers to move so that we can have like 90:100, 80:90 etc
+			si := len(dates) - 1 - (BatchSizeToDelete * batchCounter)
```

Comment:
True, I actually had that done which caused the naming to change to `SortDataMapReverse` but then took that change out since it was literally the same thing without the reverse param. I can bring it back